src/cs50.h

```diff
@@ -140,8 +168,9 @@ long long GetLongLong(void) __attribute__((deprecated));
  * upon error or no input whatsoever (i.e., just EOF). Stores string
  * on heap, but library's destructor frees memory on program's exit.
  */
-string get_string(struct prompt *p);
+
+string get_string(bool _internal, char const *fmt, ...) __attribute__((format(printf, 2, 3)));
```

Comment:
As per the macro, all calls to `get_string` will have `internal` set to false. All calls to `get_string` within the library will have `internal` set to true.

The reason we have to treat these differently because students pass variadic arguments to `get_string` the standard way (e.g. `get_string(foo, bar, baz)`). Since the rest of the `get_*` functions are themselves variadic, they can't simply forward those arguments to `get_string`. Instead they pass a `va_list` containing their arguments.

These cases need to be handled differently, and that flag indicates which case we're dealing with.